src/Internal/Service/PaymentProcessingService.php

```diff
@@ -40,12 +55,40 @@ public function __construct( StateFactory $state_factory ) {
 	 *
 	 * @throws Exception
 	 * @throws StateTransitionException In case a state cannot be initialized.
+	 * @throws PaymentRequestException  When the request is malformed. This should be converted to a failure state.
 	 */
 	public function process_payment( int $order_id ) {
-		$context = new PaymentContext();
-		$state   = $this->state_factory->create_state( InitialState::class, $context );
-		$state   = $state->process();
+		// Start with a basis context.
+		$context = $this->create_payment_context( $order_id );
+
+		// Add details from the request.
+		$request = $this->create_payment_request();
+		$request->populate_context( $context );
+
+		$state = $this->state_factory->create_state( InitialState::class, $context );
+		$state = $state->process();
```

Comment:
We are not planning to handle any exceptions upon this point?

Reply:
Not yet, or at least not here. There's an example in a different PR:

https://github.com/Automattic/woocommerce-payments/blob/a044d358cab255ce8cae5ce4a7a72e57ee586f6f/src/Internal/Payment/State/InitialState.php#L100-L104)

Right now, I'm only trying to cover the bare minimum, to allow multiple people to continue working on this. Erroneous states and exception handling will be a part of that effort.